plugins/woocommerce/includes/cli/class-wc-cli-update-command.php

```diff
@@ -84,6 +84,7 @@ public static function update() {
 		$progress->finish();
 
 		WC_Admin_Notices::remove_notice( 'update', true );
+		( new WC_Notes_Run_Db_Update() )->set_notice_actioned();
```

Comment:
How do you feel about introducing a hook for `woocommerce_db_version_updated` so we can action the notice from directly within `WC_Notes_Run_Db_Update` and keep the logic consolidated?

We can technically do this as is now with the `update_option_woocommerce_db_version` hook as well.

Reply:
Thanks for the review @joshuatf! The thing is we probably still want users to see the success message once the update has been completed in those cases where the admin started it manually. Without getting "too clever" in `WC_Notes_Run_Db_Update`, I'm not sure if we can detect whether the update was completed through the CLI or not.

That said, we _could_ add a hook specifically to the CLI update command and action the notices (from within `WC_Notes_Run_Db_Update`) by hooking onto this hook. What do you think?

Reply:
I see, so the notes are not initialized for the CLI command, correct?

In that case, I think a DB update hook for the CLI makes sense.  I'd just like to keep individual note logic consolidated to respective classes and avoid `plugins/woocommerce/includes/cli/class-wc-cli-update-command.php` becoming a dumping ground for a bunch of business logic.

Reply:
I do think ideally this would be set up so that we can fire admin code from the CLI as the current state will lead to unexpected behavior with hooks.

But for the sake of this PR, let's keep things simple 😄